sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobsCheckpointStore.cs

```diff
@@ -598,7 +600,7 @@ async Task wrapper(CancellationToken token)
         ///       "SequenceNumber":960180
         ///   }
         /// /// </remarks>
-        private static void TryReadLegacyCheckpoint(Span<byte> data, out long? offset, out long? sequenceNumber)
+        private static bool TryReadLegacyCheckpoint(Span<byte> data, out long? offset, out long? sequenceNumber)
```

Comment:
Should we allow a null offset if the sequence number is non-zero?

Reply:
What do you mean by allow? Use the sequence number if the offset is 0?

Reply:
It looks like we return `false` when an offset is null, regardless of whether there was a sequence number present.   I'm not sure we should consider that a valid case.

Reply:
>It looks like we return false when an offset is null

https://github.com/Azure/azure-sdk-for-net/pull/18059/files#diff-e78cc94b9611df5a843a75e3888d69b99158921820b2129431ebab5c2e307cfaR630

If the value is null or empty we would continue processing and return true if everything else is fine.